includes/theme-support/class-wc-twenty-twelve.php

```diff
@@ -30,8 +32,7 @@ public static function init() {
 		add_theme_support( 'wc-product-gallery-zoom' );
 		add_theme_support( 'wc-product-gallery-lightbox' );
 		add_theme_support( 'wc-product-gallery-slider' );
-		add_theme_support(
-			'woocommerce',
+		wc_get_container()->get( ThemeSupport::class )->add_default_options(
```

Comment:
Would I be right in assuming that this is intended to be used for ALL theme support cases and that we've just started the process of refactoring here? The difference in the API just feels a bit strange to me and I wonder what more general solutions you might have in mind for all of these other theme support cases as well?

Reply:
This was part of the original theme support PR in which I just changed whatever was needed to fix the bug. It might make sense to go all the way and refactor everything else if we move that part to a separate PR, but that can be discussed later.